applications/callflow/src/module/cf_ring_group.erl

```diff
@@ -193,14 +208,50 @@ order_group_members(Member, JObj) ->
          ),
     [V || {_, V} <- orddict:to_list(GroupMembers)].
 
--spec create_group_member(ne_binary(), wh_json:object(), wh_json:object()) -> wh_json:object().
-create_group_member(Key, Endpoint, Member) ->
+-spec create_group_member(ne_binary(), wh_json:object(), integer(), wh_json:object()) -> wh_json:object().
+create_group_member(Key, Endpoint, GroupWeight, Member) ->
     DefaultDelay = wh_json:get_value(<<"delay">>, Member),
     DefaultTimeout = wh_json:get_value(<<"timeout">>, Member),
     wh_json:set_values([{<<"endpoint_type">>, wh_json:get_value(<<"type">>, Endpoint)}
                         ,{<<"id">>, Key}
                         ,{<<"delay">>, wh_json:get_value(<<"delay">>, Endpoint, DefaultDelay)}
                         ,{<<"timeout">>, wh_json:get_value(<<"timeout">>, Endpoint, DefaultTimeout)}
+                        ,{<<"weight">>, GroupWeight}
                        ]
                        ,Member
                       ).
+
+-spec weighted_random_sort(wh_proplist()) -> wh_json:objects().
+weighted_random_sort(Endpoints) ->
+    random:seed(erlang:now()),
+    WeightSortedEndpoints = lists:sort(Endpoints),
+    weighted_random_sort(WeightSortedEndpoints, []).
```

Comment:
@fenollp 
I think it may be other strategies to shuffle endpoints. So this name is to know which strategy is used.